File: Data Structures & Algorithms/two-integer-sum/submission-3.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        HashMap <Integer, Integer> map = new HashMap<>();
        int[] answer = {0,0};

        for(int i = 0; i < nums.length; i++)
        {

            int current_value = nums[i];
            int difference = target - current_value;

            if(map.containsKey(difference))
            {
                answer[0] = map.get(difference);
                answer[1] = i;
            }else
            {
                map.put(current_value, i);
            }

        }

        return answer;
    }   
}

File: Data Structures & Algorithms/binary-search/submission-3.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:

        l, r = 0, len(nums)

        for i in range(len(nums)):
            
            mid = (l + r) // 2
            
            if nums[mid] == target: return mid

            l, r = (mid, r) if target > nums[mid] else (l, mid)
            
        return -1